File: frontend/src/api/datasets.ts
import { apiRequest } from "./client";

export interface SupplierRecord {
  supplier_id: number;
  supplier_name: string;
  country: string | null;
  category: string | null;
  onboarding_date: string | null;
  certification: string | null;
}

export interface ESGRecord {
  supplier_id: number;
  carbon_emission: number;
  water_usage: number;
  labor_violations: number;
  land_use_risk: string;
  esg_score: number;
}

export interface TransactionRecord {
  transaction_id: number;
  supplier_id: number;
  order_value: number;
  delivery_delay_days: number;
  defect_rate: number;
  cost_variance: number;
}

export type DatasetKey = "suppliers" | "esg" | "transactions";

export type DatasetRecord = SupplierRecord | ESGRecord | TransactionRecord;

export async function getSuppliers(): Promise<SupplierRecord[]> {
  return apiRequest<SupplierRecord[]>("/suppliers");
}

export async function getEsgRecords(): Promise<ESGRecord[]> {
  return apiRequest<ESGRecord[]>("/esg");
}

export async function getTransactions(): Promise<TransactionRecord[]> {
  return apiRequest<TransactionRecord[]>("/transactions");
}

File: backend/app/schemas/datasets.py
from pydantic import BaseModel, ConfigDict


class SupplierRecord(BaseModel):
    supplier_id: int
    supplier_name: str
    country: str | None = None
    category: str | None = None
    onboarding_date: str | None = None
    certification: str | None = None

    model_config = ConfigDict(from_attributes=True)


class ESGRecord(BaseModel):
    supplier_id: int
    carbon_emission: float
    water_usage: float
    labor_violations: float
    land_use_risk: str | None = None
    esg_score: float

    model_config = ConfigDict(from_attributes=True)


class TransactionRecord(BaseModel):
    transaction_id: int
    supplier_id: int
    order_value: float
    delivery_delay_days: float
    defect_rate: float
    cost_variance: float

    model_config = ConfigDict(from_attributes=True)


class SupplierPerformanceRecord(BaseModel):
    supplier_id: int
    avg_delay: float
    avg_defect: float
    avg_cost_variance: float
    risk_score: float

    model_config = ConfigDict(from_attributes=True)
